pkg/grafana/provider.go

```diff
@@ -105,6 +105,15 @@ func (p *Provider) GetHandlers() []grizzly.Handler {
 }
 
 func (p *Provider) SetupProxy() (*httputil.ReverseProxy, error) {
+	client, err := p.Client()
+	if err != nil {
+		return nil, err
+	}
+	_, err = client.Dashboards.GetHomeDashboard()
```

Comment:
I understand that it's checking any kind of error here. I mean, at least that we want to fail anyway its ok but the error returned should be different.

If we want to specify that the error is because the authentication is invalid, we should cast the error with [this](https://github.com/grafana/grafana-openapi-client-go/blob/main/client/dashboards/get_home_dashboard_responses.go#L33-L38).

So in that case should be something like:
```go
if err != nil {
  if errors.Is(err, &dashboards.GetHomeDashboardUnauthorized{}) {
    return nil, fmt.Errorf("error checking authentication: %v", err)
  }
  return nil, fmt.Errorf("error setting the proxy: %v", err)
}
```

Reply:
applied